{"metadata":{"kernelspec":{"language":"python","display_name":"Python 3","name":"python3"},"language_info":{"name":"python","version":"3.7.12","mimetype":"text/x-python","codemirror_mode":{"name":"ipython","version":3},"pygments_lexer":"ipython3","nbconvert_exporter":"python","file_extension":".py"}},"nbformat_minor":4,"nbformat":4,"cells":[{"cell_type":"code","source":"!python /kaggle/input/ayanan/train.py --optimizer adam --epochs 5 --learning_rate 0.001 --batch_size 32 --loss squared_error --activation tanh --num_layers 5 --hidden_size 32 --weight_init xavier","metadata":{"execution":{"iopub.status.busy":"2023-03-12T19:05:42.032838Z","iopub.execute_input":"2023-03-12T19:05:42.033322Z","iopub.status.idle":"2023-03-12T19:07:19.983112Z","shell.execute_reply.started":"2023-03-12T19:05:42.033277Z","shell.execute_reply":"2023-03-12T19:07:19.981315Z"},"trusted":true},"execution_count":3,"outputs":[{"name":"stdout","text":"\u001b[34m\u001b[1mwandb\u001b[0m: Currently logged in as: \u001b[33mcs22m025\u001b[0m. Use \u001b[1m`wandb login --relogin`\u001b[0m to force relogin\n\u001b[34m\u001b[1mwandb\u001b[0m: Appending key for api.wandb.ai to your netrc file: /root/.netrc\nDownloading data from https://storage.googleapis.com/tensorflow/tf-keras-datasets/train-labels-idx1-ubyte.gz\n29515/29515 [==============================] - 0s 0us/step\nDownloading data from https://storage.googleapis.com/tensorflow/tf-keras-datasets/train-images-idx3-ubyte.gz\n26421880/26421880 [==============================] - 1s 0us/step\nDownloading data from https://storage.googleapis.com/tensorflow/tf-keras-datasets/t10k-labels-idx1-ubyte.gz\n5148/5148 [==============================] - 0s 0us/step\nDownloading data from https://storage.googleapis.com/tensorflow/tf-keras-datasets/t10k-images-idx3-ubyte.gz\n4422102/4422102 [==============================] - 0s 0us/step\n\u001b[34m\u001b[1mwandb\u001b[0m: wandb version 0.13.11 is available!  To upgrade, please run:\n\u001b[34m\u001b[1mwandb\u001b[0m:  $ pip install wandb --upgrade\n\u001b[34m\u001b[1mwandb\u001b[0m: Tracking run with wandb version 0.13.10\n\u001b[34m\u001b[1mwandb\u001b[0m: Run data is saved locally in \u001b[35m\u001b[1m/kaggle/working/wandb/run-20230312_190601-mou2czhg\u001b[0m\n\u001b[34m\u001b[1mwandb\u001b[0m: Run \u001b[1m`wandb offline`\u001b[0m to turn off syncing.\n\u001b[34m\u001b[1mwandb\u001b[0m: Syncing run \u001b[33miconic-thunder-452\u001b[0m\n\u001b[34m\u001b[1mwandb\u001b[0m: ‚≠êÔ∏è View project at \u001b[34m\u001b[4mhttps://wandb.ai/cs22m025/assignment1\u001b[0m\n\u001b[34m\u001b[1mwandb\u001b[0m: üöÄ View run at \u001b[34m\u001b[4mhttps://wandb.ai/cs22m025/assignment1/runs/mou2czhg\u001b[0m\n___________________________________________________________________\nRunning Adam.....\nepoch: 1 loss: 0.5809147121779843 0.5910253524153617\nepoch: 2 loss: 0.5230797265506246 0.5315270351232648\nepoch: 3 loss: 0.49756387267997215 0.5010591476130467\nepoch: 4 loss: 0.47920526386724177 0.48156573055077995\nepoch: 5 loss: 0.46488899964801156 0.4694015986638166\n------------------------------------------------------------------\nAccuracy on training data: 83.9037037037037\nAccuracy on validation data: 83.68333333333334\n\u001b[34m\u001b[1mwandb\u001b[0m: Waiting for W&B process to finish... \u001b[32m(success).\u001b[0m\n\u001b[34m\u001b[1mwandb\u001b[0m: \\ 0.001 MB of 0.039 MB uploaded (0.000 MB deduped)\n\u001b[34m\u001b[1mwandb\u001b[0m: Run history:\n\u001b[34m\u001b[1mwandb\u001b[0m:               epoch ‚ñÅ‚ñÅ‚ñÅ‚ñÅ‚ñÅ\n\u001b[34m\u001b[1mwandb\u001b[0m:        training_acc ‚ñÅ‚ñÖ‚ñÜ‚ñá‚ñà\n\u001b[34m\u001b[1mwandb\u001b[0m:       training_loss ‚ñà‚ñÖ‚ñÉ‚ñÇ‚ñÅ\n\u001b[34m\u001b[1mwandb\u001b[0m:     validation cost ‚ñà‚ñÖ‚ñÉ‚ñÇ‚ñÅ\n\u001b[34m\u001b[1mwandb\u001b[0m: validation_accuracy ‚ñÅ‚ñÑ‚ñÜ‚ñá‚ñà\n\u001b[34m\u001b[1mwandb\u001b[0m: \n\u001b[34m\u001b[1mwandb\u001b[0m: Run summary:\n\u001b[34m\u001b[1mwandb\u001b[0m:               epoch 5\n\u001b[34m\u001b[1mwandb\u001b[0m:        training_acc 83.9037\n\u001b[34m\u001b[1mwandb\u001b[0m:       training_loss 0.46489\n\u001b[34m\u001b[1mwandb\u001b[0m:     validation cost 0.4694\n\u001b[34m\u001b[1mwandb\u001b[0m: validation_accuracy 83.68333\n\u001b[34m\u001b[1mwandb\u001b[0m: \n\u001b[34m\u001b[1mwandb\u001b[0m: üöÄ View run \u001b[33miconic-thunder-452\u001b[0m at: \u001b[34m\u001b[4mhttps://wandb.ai/cs22m025/assignment1/runs/mou2czhg\u001b[0m\n\u001b[34m\u001b[1mwandb\u001b[0m: Synced 6 W&B file(s), 0 media file(s), 0 artifact file(s) and 0 other file(s)\n\u001b[34m\u001b[1mwandb\u001b[0m: Find logs at: \u001b[35m\u001b[1m./wandb/run-20230312_190601-mou2czhg/logs\u001b[0m\n","output_type":"stream"}]},{"cell_type":"markdown","source":"!python /kaggle/input/xyzwuv/train.py --optimizer adam --epochs 5 --l_rate 0.001 --batch_size 32 --loss squared_error --activation tanh --n_hlayers 5 --hlayer_size 32 --initializer xavier","metadata":{"execution":{"iopub.status.busy":"2023-03-11T16:56:54.719853Z","iopub.execute_input":"2023-03-11T16:56:54.720275Z"}}},{"cell_type":"markdown","source":"!python /kaggle/input/xyzwuv/train.py --optimizer adam --epochs 5 --l_rate 0.001 --batch_size 32 --loss squared_error --activation tanh --n_hlayers 5 --hlayer_size 32 --initializer xavier","metadata":{"execution":{"iopub.status.busy":"2023-03-11T16:57:42.460968Z","iopub.execute_input":"2023-03-11T16:57:42.461391Z"}}},{"cell_type":"code","source":"Loss=[0,0,0,0]\nepoch=[1,2,3,4]\n!python /kaggle/input/training/original.py plotloss(Loss,epoch)","metadata":{"execution":{"iopub.status.busy":"2023-03-10T16:55:15.478134Z","iopub.execute_input":"2023-03-10T16:55:15.478694Z","iopub.status.idle":"2023-03-10T16:55:16.521989Z","shell.execute_reply.started":"2023-03-10T16:55:15.478640Z","shell.execute_reply":"2023-03-10T16:55:16.520394Z"},"trusted":true},"execution_count":13,"outputs":[{"name":"stdout","text":"/bin/bash: -c: line 0: syntax error near unexpected token `('\n/bin/bash: -c: line 0: `python /kaggle/input/training/original.py plotloss(Loss,epoch)'\n","output_type":"stream"}]},{"cell_type":"code","source":"import original","metadata":{"execution":{"iopub.status.busy":"2023-03-10T16:56:36.853581Z","iopub.execute_input":"2023-03-10T16:56:36.854075Z","iopub.status.idle":"2023-03-10T16:56:36.881625Z","shell.execute_reply.started":"2023-03-10T16:56:36.854027Z","shell.execute_reply":"2023-03-10T16:56:36.879452Z"},"trusted":true},"execution_count":14,"outputs":[{"traceback":["\u001b[0;31m---------------------------------------------------------------------------\u001b[0m","\u001b[0;31mModuleNotFoundError\u001b[0m                       Traceback (most recent call last)","\u001b[0;32m/tmp/ipykernel_28/554201614.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0;32mimport\u001b[0m \u001b[0moriginal\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m","\u001b[0;31mModuleNotFoundError\u001b[0m: No module named 'original'"],"ename":"ModuleNotFoundError","evalue":"No module named 'original'","output_type":"error"}]}]}